### PR TITLE
Setup Players, Game and Message classes

### PR DESCRIPTION
**Pull request by timonrey**
The code includes a new game class, where the game can be activated. 
The buttonWasPressedWithValue method will be triggered by the button
In the Main class, we have new instances of the Players, Game and Message class.
The methods in the new Message class can tell you the information of the button push.
The Players class is going to change the score of each player after a button push.

### CODE REVIEW

**Comment by PhilippSpo** [comment]
> The Players class is going to change the score of each player after a button push.

The Players class will not change any value. What will instead?

File: src/Game.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+public class Game {
+
+    Players playerOne;
+    Players playerTwo;
+    Players scorePlayerOne = new Players();
```
Comment (PhilippSpo) [inline src/Game.java]: This is still the old code. I guess you have forgot to commit 😉

**Review by PhilippSpo — CHANGES_REQUESTED** [review]

**Comment by timonrey** [comment]
Can i make a new pull request? i have changed and deleted some names and comments…? 
Would that make sense?

File: src/Game.java (hunk 1 above)
Comment (emmenko) [reply src/Game.java]: Should I wait before reviewing this?

**Comment by PhilippSpo** [comment]
Just commit the stuff on this branch to update the PR

**Comment by timonrey** [comment]
Instead of the Player class, the methods in this class will change the score value of the player who pushed the button

**Comment by emmenko** [comment]
@TimonRey don't answer from the Email, come here in Github and answer in the website. Thanks ;)

**Comment by timonrey** [comment]
@PhilippSpo Instead of the Player class, the methods in this class will change the score value of the player who pushed the button

File: src/Game.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+public class Game {
+
+    Players playerOne;
+    Players playerTwo;
+
+    public Game(Players firstPlayerName, Players secondPlayerName) {
```
Comment (emmenko) [inline src/Game.java]: Why is it called `xxxName`?

File: src/Game.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        this.playerOne = firstPlayerName;
+        this.playerTwo = secondPlayerName;
+        
+    }
+
+    public void buttonWasPressedWithValue(Message mail) {
```
Comment (emmenko) [inline src/Game.java]: Nitpick: we should probably rename this method. Also why is the variable called `mail`? 😄

File: src/Main.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        Players philipp = new Players();
+        Game match = new Game(nicola, philipp);
+        Message nicolaScores = new Message(nicola, 1);
+        Message philippScores = new Message(philipp, 1);
+        Message nicolaReset = new Message(nicola, 3);
+        Message philippReset = new Message(philipp,3);
+
+       // match.playerTwo = new Players()
+
+        match.buttonWasPressedWithValue(nicolaScores);
```
Comment (emmenko) [inline src/Main.java]: Wouldn't be better to create a new message each time you execute this function?

`match.buttonWasPressedWithValue(new Message(...))`

File: src/Message.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+public class Message {
+
+    Players player;
+    int actionType;
+
+    public Message (Players playerNumber, int actionType) {
+        this.player = playerNumber;
+        this.actionType = actionType;
+    }
```
Comment (emmenko) [inline src/Message.java]: The `Message` should look like this right?

https://github.com/TimonRey/PingPongApp/blob/89ee8306d258e9b731add35f5c784c8258fe3749/src/main/java/PingPong.java#L48-L51

So what you get from a `Message` is
* `buttonId`: it's a string
* `actionType`: also a string

File: src/Players.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    public void sub () {
+        points = points - 1;
+    }
+
+    public void reset () {
+        points = 0;
+    }
+
+
+    public int getter () {
```
Comment (emmenko) [inline src/Players.java]: Maybe we can rename this to something like `getScore` or `getPoints`, `getter` is a bit generic

File: src/Players.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    public Players() {
+
+    }
+
+    public void add () {
+        points = points +1;
+    }
+
+    public void sub () {
+        points = points - 1;
```
Comment (emmenko) [inline src/Players.java]: Nitpick: you should probably don't allow negative numbers

**Review by emmenko — DISMISSED** [review]
Good work so far! 🙌

I left some initial comments, mostly related to naming ;)

File: src/Game.java (hunk 2 above)
Comment (timonrey) [reply src/Game.java]: It makes it easier for me to understand what the __constructor__ will *accept*

Comment (PhilippSpo) [reply src/Game.java]: But the constructor won't accept player **names**. It accepts the whole **player** itself, not only the name.

File: src/Game.java (hunk 3 above)
Comment (timonrey) [reply src/Game.java]: You're right, the name is too long. I don't know why but the name before was more ___stupid___ :)

File: src/Main.java (hunk 4 above)
Comment (timonrey) [reply src/Main.java]: That's included in the ___new version___, i think

File: src/Message.java (hunk 5 above)
Comment (timonrey) [reply src/Message.java]: No, the __button__ variable is a *string* but the __actionType__ is an *int*...just a number

File: src/Players.java (hunk 7 above)
Comment (timonrey) [reply src/Players.java]: oh...you're right! But the __sub method__ will be changed into a __set method__ in the next step

File: src/Players.java (hunk 6 above)
Comment (timonrey) [reply src/Players.java]: # Sure!
 it was just practically for me to __understand__ what a getter method is

**Review by PhilippSpo — APPROVED** [review]

**Pull request merged**